misaligned.cpp

```diff
@@ -1,21 +1,50 @@
 #include <assert.h>
 #include <iostream>
+#include <sstream>
+#include <string>
 
-int printColorMap() {
+int printColorMap(std::ostream& out) {
     const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
     const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
     for (i = 0; i < 5; i++) {
         for (j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            out << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
         }
     }
     return i * j;
 }
 
+// Test
+int testPrintColorMap() {
+    int num = 0;
+    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+    int i = 0, j = 0;
+    std::stringstream ss;
+    printColorMap(ss);
+    std::string input;
+    while (std::getline(ss, input, '\n')) {
+        std::string expectedInputStr = std::to_string(i * 5 + j) +

```

Comment:
Let the expected reflect what the user is expecting. The user expects the manual to have pair numbers from 1 to 25, but the 'expected' here goes from 0 to 24